aggregate/housing_security/nycha_tenants.py

```diff
@@ -0,0 +1,81 @@
+from aggregate.decennial_census.decennial_census_001020 import decennial_census_001020
+from aggregate.aggregation_helpers import order_aggregated_columns
+import pandas as pd
+from internal_review.set_internal_review_file import set_internal_review_files
+from utils.PUMA_helpers import clean_PUMAs, puma_to_borough
+dcp_pop_races = ["anh", "bnh", "hsp", "onh", "wnh"]
+
+
+race_labels = ["", "_wnh", "_bnh", "_hsp", "_anh", "_onh"]
+
+def nycha_tenants(geography: str, write_to_internal_review=False):
+    assert geography in ["citywide", "borough", "puma"]
+
+    clean_data = load_clean_nycha_data()
+    census20 = decennial_census_001020(geography=geography, year="1519")
+    if geography == "puma":
+        final = get_percentage(pd.concat([census20, clean_data], axis=1))
+    elif geography == "borough":
+        agg = clean_data.groupby("borough").agg("sum")
+        census20_agg = pd.concat([census20, agg], axis=1)
+        final = get_percentage(census20_agg)
+    elif geography == "citywide":
+        agg = clean_data.groupby("citywide").agg("sum")
+        census20_agg = pd.concat([census20, agg], axis=1)
+        final = get_percentage(census20_agg)
+        
+    final = final.round(2)
+
+    order_cols = order_aggregated_columns(
+        df=final,
+        indicators_denom=[("pop",), ("nycha_tenants",)],
+        categories={"pop": ["pop"], "nycha_tenants": ["nycha_tenants"], "race": dcp_pop_races},
+        return_col_order=True,
+        exclude_denom=True
+    )
+    # 24 final columns match the results 
+    final_cols = [col for col in order_cols if "cv" not in col]
+    final_cols = [col for col in final_cols if "moe" not in col]
+    final = final.reindex(columns=final_cols)
+
+    if write_to_internal_review:
+        set_internal_review_files(
+            [(final, "nycha_tenants.csv", geography)],
+            "housing_security",
+        )
+
+    return final 
+
+def load_clean_nycha_data():
+
+    census20 = decennial_census_001020("puma", year="1519") # this is in fact pulling 2020 census but design has it mapped from 1519 to 2020
+
+    read_excel_arg = {
+        "io": "resources/housing_security/Equitable.Development.Data.Tool.-.Displacement.Risk.Index.2-8-2022.1.xlsx",
+        "sheet_name": "PUMA",
+        "usecols": "A, F:Q",
+        "nrows": 41,
+        "dtype": float
+    }
+    nycha_data = pd.read_excel(**read_excel_arg)
+    nycha_data.rename(columns={"PUMA (2010)": "puma"}, inplace=True)
+    nycha_data.puma = nycha_data.puma.apply(func=clean_PUMAs)
+    nycha_data["borough"] = nycha_data.apply(axis=1, func=puma_to_borough)
+    nycha_data["citywide"] = "citywide"
+    nycha_data.set_index("puma", inplace=True)
+    # calculating the total for each race categories
+    for i in range(6):
```

Comment:
I am anxious about iterating over numeric range here. Fragile design that works when run on new source data but could be pointing at wrong columns. Better to iterate through races and then if source data changes the code breaks

Reply:
Hey @SashaWeinstein  don't totally disagree about the enumerating over index vs. column name but I need to think about how this might complicate the design. I resorted to this based on the assumptions that 1. this would be one-off for this year and also we need to get this out in a pressing deadline